test/e2e/apimachinery/aggregator.go

```diff
@@ -72,13 +72,35 @@ var _ = SIGDescribe("Aggregator", func() {
 		aggrclient = f.AggregatorClient
 	})
 
-	It("Should be able to support the 1.10 Sample API Server using the current Aggregator", func() {
+	// We have to put the version skew test and the storage version hash
+	// test in the same "It" block, because
+	// 1. These two tests require different versions of the sample
+	// apiserver. Otherwise we would be able to setup the server in a
+	// "BeforeEach" block, and run the tests in two different "It" blocks.
+	// 2. We cannot install two sample apiservers at the same time, because
+	// they both serve at "wardle.k8s.io/v1alpha", which is hard-coded.
+	// Otherwise we would be able to run the two tests in two "It" blocks
+	// in parallel.
+	// 3. ginkgo cannot enforce the order of two "It" blocks, unless all
+	// tests in the test suite are serial.  Otherwise we would be able to
+	// run the tests in two "It" blocks in serial.
+	It("Should be able to support the 1.10 Sample API Server using the current Aggregator; Should support StorageVersionHash in the 1.14 Sample API Server", func() {
 		// Make sure the relevant provider supports Agggregator
 		framework.SkipUnlessServerVersionGTE(serverAggregatorVersion, f.ClientSet.Discovery())
-		framework.SkipUnlessProviderIs("gce", "gke")
+		framework.SkipUnlessProviderIs("gce", "gke", "local")
 
 		// Testing a 1.10 version of the sample-apiserver
-		TestSampleAPIServer(f, imageutils.GetE2EImage(imageutils.APIServer))
+		setupSampleAPIServer(f, imageutils.GetE2EImage(imageutils.OldAPIServer))
+		testOldSampleAPIServer(f)
+		cleanTest(c, aggrclient, ns)
+
+		// Aggregated apiserver has full control over its own discovery
+		// document, so even if the StorageVersionHash feature flag is
+		// disabled, the 1.14 sample apiserver still serves the storage
+		// version hash field and this test still passes.
```

Comment:
it seems odd to test an alpha feature with an alpha option in a non-alpha CI test. is there a reason not to put this in a separate `It` block with a `[Feature:...]` tag?

Reply:
also, does that mean the sample apiserver unconditionally enables this new alpha feature in the 1.14 image? I wouldn't have expected our examples to enable alpha features by default

Reply:
For why not running in two `It` blocks, see the long [comment](https://github.com/kubernetes/kubernetes/pull/72942/files#diff-c944d1288edcaf37beebab811603bfd8R75) I added at the existing `It` block. 

Though not ideal, if we can move the sample apiserver tests to serial tests, we can split them into multiple `It` blocks.

> also, does that mean the sample apiserver unconditionally enables this new alpha feature in the 1.14 image?

I removed the comment starting from line 97, it's wrong. The alpha feature was disabled by default in sample apiserver.

Reply:
I kept the two tests in a single `It` block. Unless we refactor the sample-apiserver to dynamically register resource endpoints, we can't realistically make this test better.